servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpServerConfig.java

```diff
@@ -36,10 +43,54 @@ HttpConfig httpConfig() {
     }
 
     ReadOnlyHttpServerConfig asReadOnly() {
+        applySslConfigOverrides();
         final ReadOnlyHttpServerConfig roConfig = new ReadOnlyHttpServerConfig(this);
         if (roConfig.tcpConfig().sslContext() == null && roConfig.h1Config() != null && roConfig.h2Config() != null) {
             throw new IllegalStateException("Cleartext HTTP/1.1 -> HTTP/2 (h2c) upgrade is not supported");
         }
         return roConfig;
     }
+
+    private void applySslConfigOverrides() {
+        ServerSslConfig sslConfig = tcpConfig.sslConfig();
+        if (sslConfig != null) {
+            sslConfig = new DelegatingHttpServerSslConfig(sslConfig,
+                    httpAlpnProtocols(sslConfig.alpnProtocols(), httpConfig.supportedAlpnProtocols()));
+            Map<String, ServerSslConfig> sniMap = tcpConfig.sniConfig();
+            if (sniMap == null) {
+                tcpConfig.sslConfig(sslConfig);
+            } else {
+                // Make a copy in case the original map is unmodifiable.
+                Map<String, ServerSslConfig> sniMapOverrides = new LinkedHashMap<>(sniMap.size());
```

Comment:
Presumably uses LinkedHashMap for iteration order but that is not the type of sniMapOverrides or sniMap. Why is it a LinkedHashMap?

Reply:
Yes LHM is to preserve iteration order. I'll add a comment.

Reply:
Perhaps sniMap should be a LinkedHashMap as well for the same reason?

Reply:
I don't think we need to restrict which map the user is able to provide. If iteration ordering is important they can always provide LHM, but maybe there is no overlap/wildcards and they don't need the additional overhead.